src/DynamicData/List/ObservableListEx.cs

```diff
@@ -2254,6 +2254,13 @@ public static IObservable<IChangeSet<T>> WhereReasonsAreNot<T>(this IObservable<
             throw new ArgumentException("Must enter at least 1 reason", nameof(reasons));
         }
 
+        // Optimization that avoids removing the indexes if only refreshes are removed
+        if (reasons.Length == 1 && reasons[0] == ListChangeReason.Refresh)
+        {
+            // Caller probably meant to call this instead, so help them out.
+            return source.SuppressRefresh().NotEmpty();
```

Comment:
Actually, I just look up SuppressRefresh() and it overloads WhereReasonsAreNot().  This means we'll have to use some specialized handling in this method.

Reply:
The Cache version does that, but not the List version which explicitly strips out the Refresh events.

If you'd prefer, I can change the SuppressRefresh to match the Cache version and move the old code to be inside of `WhereReasonsAreNot`.  Either way, the logic will appear in only one place.

I'm not sure which approach is better, but maybe it would be good to align the List and Cache versions.

Reply:
I like having the implementation in `SuppressRefresh` better because then it avoids the allocation of the ListChangeReason[] and then checking it for the one single value.  It's like the more optimized version.  Doing it the other way means there is no optimized version, just an easier way to call it.

But it does mean not bringing the List/Cache versions back together...  But I think that's okay because they don't need to be aligned.

Reply:
Okay, I changed my mind.  I put the logic for both cases (Refresh only and everything else) in the `WhereReasonsAreNot` operator.  Yes, there's no way to call it without the array, but all of the logic is in one place instead of two.

Plus, that enabled me to change the `SuppressRefresh` operator to invoke `WhereReasonsAreNot` so that it is the same as the Cache version.

Both of these simplify the implementation which is a good thing for maintainers.